docs: add section for custom image registry options (#2386)
spectrocloud/librarium: docs/docs-content/clusters/edge/site-deployment/deploy-custom-registries/deploy-external-registry.md

---
sidebar_label: "Deploy Cluster with a Private External Registry"
title: "Deploy Cluster with a Private External Registry"
description: "Instructions for deploying an Edge cluster with a private external registry."
hide_table_of_contents: false
sidebar_position: 60
tags: ["edge"]
---

Palette Edge provides support for downloading images from authenticated external registries. You can instruct the
Palette agent to download images from an authenticated external registry by specifying the address and the credentials
for the registry in the user data used to build your Edge Installer ISO.

Once you specify an external registry, images for all elements of the cluster are expected to be in the external
registry. This includes the provider images, images for the network and storage layer, and images for all application
layers. All images specified in the cluster profile will have their registry URL replaced by the registry URL of the
external image registry.

:::tip

You can use a private external registry together with a local Harbor image registry. All images from the external

> private registry will be stored in the local Harbor registry after the initial download, which allows you to reduce

Reviewer: ```suggestion
private registry will be stored in the local Harbor registry after the initial download, which allows you to reduce the
```

> bandwidth use and protect against outages. For more information, refer to

Reviewer: ```suggestion
bandwidth use and gain protection against outages. For more information, refer to
```

Reviewer: Or maybe protect your system or something? Protect feels a bit out of place here

[Enable Local Harbor Registry](./local-registry.md).

:::

## Limitations

- Palette Edge supports basic username/password authentication. Token authentication schemes used by services such as
  AWS ECR and Google Artifact Registry are not supported.

## Prerequisites

- Specifying the external registry and providing credentials happens during the EdgeForge process. You should become
  familiar with EdgeForge before following this guide. Refer to
  [Build Edge Artifacts](../../edgeforge-workflow/palette-canvos/palette-canvos.md) to learn how to build Edge Installer
  ISO and provider images.

> - A physical or virtual Linux machine with _AMD64_ (also known as _x86_64_) processor architecture to build the Edge

Reviewer: Why do we need italic here?  _AMD64_ (also known as _x86_64_)

  artifacts. You can issue the following command in the terminal to check your processor architecture.

  ```bash
  uname -m
  ```

- Minimum hardware configuration of the Linux machine:

  - 4 CPU
  - 8 GB memory
  - 50 GB storage

- [Git](https://cli.github.com/manual/installation). You can ensure git installation by issuing the `git --version`
  command.

- [Docker Engine](https://docs.docker.com/engine/install/) version 18.09.x or later. You can use the `docker --version`

>   command to view the existing Docker version. You should have root-level or `sudo` privileges on your Linux machine to
>   create privileged containers.

Reviewer: ```suggestion
  command to view the existing Docker version.
  
  - You should have root-level or `sudo` privileges on your Linux machine to
  create privileged containers.
```

Reviewer: I think it makes more sense for this to be its own prereq
